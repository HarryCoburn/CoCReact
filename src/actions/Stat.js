import * as StatMsg from "./messages/statMsg";

/**
 * Sends message to set a stat to a particular number
 * Payload shape: { strength: 23 }
 * @param {object} payload
 * @return {object} Redux action
 */
export function _setStats(payload) {
  if (!(payload instanceof Object) || payload === undefined) {
    throw Error("Core._setStats did not receive an object");
  }
  return {
    type: StatMsg.SET_STATS,
    payload
  };
}

/**
 * Sends message to increase or decrease a stat by a particular number
 * Payload shape: { strength: 8 } or { strength: -19 }
 * @param {object} payload
 * @return {object} Redux action
 */
export function _statChange(payload) {
  if (!(payload instanceof Object) || payload === undefined) {
    throw Error("Core._statChange did not receive an object");
  }
  return {
    type: StatMsg.CHANGE_STATS,
    payload
  };
}

export const _hideStats = {
  type: StatMsg.HIDE_STATS
};

export const _showStats = {
  type: StatMsg.SHOW_STATS
};
