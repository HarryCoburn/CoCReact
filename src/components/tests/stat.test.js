import React from "react";
import renderer from "react-test-renderer";
import Stat from "../stat";

const dummy = {
  stat: "strength",
  name: "Strength",
  value: 50,
  max: 100
};

describe("Stat Component", () => {
  xit("renders correctly", () => {
    const component = renderer.create(
      <Stat
        key={dummy.stat}
        id={dummy.stat}
        name={dummy.name}
        value={dummy.value}
        max={dummy.max}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
