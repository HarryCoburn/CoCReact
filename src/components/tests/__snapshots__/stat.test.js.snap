// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Component renders correctly 1`] = `
<div
  className="statFrame"
>
  <div
    className="statGradient"
  >
    <p
      className="statLabel"
    >
      Strength
      : 
      50
       
    </p>
  </div>
  <div
    className="statUpDown"
  />
</div>
`;
